### ISSUE TITLE
Add new custom page option for handle404 

### ISSUE THREAD

**Issue by daniloff200** [scullyio/scully#1287]
### Description

Right now we have `index` and `404` options, which allows us to show pages for unhandled routes from **./dist** folder .

I have a case when I have some private routes, protected by AuthGuard, which ignored by Scully and I want to show **index.html** or  some custom page, like, some loading page, and it's not located in `./dist` folder.

OR
Another case:
We just deploy the ` ./dist/static` folder,  and don't want to deploy the whole content of ` ./dist` folder to have ability to show `index.html`, which available by using **`index`** option.

### Describe the solution you'd like

A new option -> `customPage`. 
We can put there a path to some .html  page, which we want to show for ignored/private routes. 
Also, in general, it can be the same index.html or 404.html, but, not located in` ./dist` folder, but, in other places

**Comment by SanderElias** [comment]
@daniloff200 Once deployed to production, how is the app getting the page?

**Comment by daniloff200** [comment]
Not sure, that I undersand your question correct, but...

Right now I'm showing **index.html** for my private routes, that ignored in Scully config,
 from my app/src build, and, it's located in dist root folder (I copy it there during the deployment process)

There's how it looks. Static folder - is a Scully build.
 ![Selection_020](https://user-images.githubusercontent.com/13692220/110196636-7fb43f00-7e4e-11eb-9ac4-8a4bafe9c0b1.png)
 
And right now I have to deploy whole /dist folder, to show proper html for these private routes, but, in general, I would like to deploy only /static folder. I've already tried it -> all pages in app work correct, except private routes, when I reload the page or, going there by direct link

**Comment by SanderElias** [comment]
So, you want your server to serve a 404 when the route isn't in your Angular app?
That is a setting you need to do on your production server. Scully can't generate pages that are not there.
If you are using Scully, it will render all the pages you tell it to render. But pages that are not generated are _not_ there, so you can't provide functionality on something that you don't provide.
When you build with Scully, there is a 404.html in your static folder, that is your unmodified application. Let your server serve that as 404.
Alternatively, you can actually create a 404 page in your app, and then teach your server to use that page if an URL doesn't resolve to a file.

**Comment by daniloff200** [comment]
So, basically, I ended up with another solution.

My initial task was to avoid behaviour, like `reload the page -> login page -> user profile`, which happened for some routes in the app, which are under guard (in my case it was `AuthGuard`), like **UserProfile, Checkout**, etc.
It happens, because auth guard routes to login / landing page during the render, and in this case my idea was to show something, like spinner page, and not 404 or index.html, which are available as options for `handle404` now.

But, my initial solution was not the best one.

In this case it's better to use `isScullyRunning()` and render something alternative page in this case.
I used it on my login page, when user redirected from the routes under guard, and it was like:

`this.showLoadingSpinner = isScullyRunning();`

and according to `this.showLoadingSpinner` user can see the loading spinner, and then redirected back to the proper page, when it becomes active
